Source/WebCore/dom/Document.cpp

```diff
@@ -836,6 +836,9 @@ Document::~Document()
     if (hasRareData())
         clearRareData();
 
+    if (auto completionHandler = std::exchange(m_whenWindowLoadEventOrDestroyed, nullptr))
+        completionHandler();
```

Comment:
optional: Just calling the completion handler does the exchange with `nullptr` by the way, so this would have been equivalent:

```suggestion
    if (m_whenWindowLoadEventOrDestroyed)
        m_whenWindowLoadEventOrDestroyed();
```